### ISSUE TITLE
Training takes a lot of time and gets loss as NaN at one point

### ISSUE THREAD

**Issue by samin9796** [facebookresearch/wav2letter#444]
I am using Nvidia geforce gtx 1080Ti 7 gpu to train 240 hours of data. It took around 10 hours and epoch 1 was still going on. Using the nvidia-smi, I found that all the gpu were not fully utilized. At one point while training, loss function increased exponentially. Here is the train.cfg:

--runname=sust_full
--rundir=/data/ahnaf/wav2letter/dataset_prep/
--datadir=/data/ahnaf/wav2letter/dataset_prep/
--tokensdir=/data/ahnaf/wav2letter/dataset_prep/
--train=train.lst
--valid=validation.lst
--lexicon=/data/ahnaf/wav2letter/dataset_prep/lexicon.txt
--input=wav
--tokens=tokens.txt
--lr=0.1
--maxgradnorm=1.0
--momentum=0.9
--mfsc=true
--batchsize=5
--iter=70
--lrcrit=0.0002
--archdir=/data/ahnaf/wav2letter/dataset_prep/
--arch=network.arch
--enable_distributed=true
--criterion=ctc

Here is the log file:
[001_log.zip](https://github.com/facebookresearch/wav2letter/files/3854525/001_log.zip)

How can I make the training faster and how to avoid loss NaN error?

**Comment by tlikhomanenko** [comment]
Hi @samin9796,

It seems you are using `reportiters=1`, is it true? I See from your log that runtime is 0s so the main time is in evaluation on validation set which is called after each updates if you run with `reportiters=1`. Please use larger value, for example `reportiters=500`.

For convergence issue you could try to decrease `maxgradnorm`, `lr`.

**Comment by samin9796** [comment]
Thank you @tlikhomanenko . I was using reportiters=1. Now training is much faster. However, I trained twice, first time was 70 epochs and the second time was 200 epochs. lr and other parameters were different at those two training. I am getting roughly 58 as wer and 17 as ter. Using deepspeech2, I got wer=26 and cer=7. I am expecting similar or better result from wav2letter. I was using the same network architecture provided for librispeech to train bangla 240 hours of data. Can you please provide some insight so that it converges?
First time:
train.cfg:
--runname=sust_full
--rundir=/data/ahnaf/wav2letter/dataset_prep/
--datadir=/data/ahnaf/wav2letter/dataset_prep/
--tokensdir=/data/ahnaf/wav2letter/dataset_prep/
--train=train.lst
--valid=validation.lst
--lexicon=/data/ahnaf/wav2letter/dataset_prep/lexicon.txt
--input=wav
--tokens=tokens.txt
--lr=0.05
--maxgradnorm=0.7
--momentum=0.9
--mfsc=true
--batchsize=20
--iter=70
--lrcrit=0.0002
--archdir=/data/ahnaf/wav2letter/dataset_prep/
--arch=network.arch
--enable_distributed=true
--criterion=ctc
log file: 
[001_log_70.zip](https://github.com/facebookresearch/wav2letter/files/3883659/001_log_70.zip)

Second time:
train.cfg:
--runname=sust_full
--rundir=/data/ahnaf/wav2letter/dataset_prep/
--datadir=/data/ahnaf/wav2letter/dataset_prep/
--tokensdir=/data/ahnaf/wav2letter/dataset_prep/
--train=train.lst
--valid=validation.lst
--lexicon=/data/ahnaf/wav2letter/dataset_prep/lexicon.txt
--input=wav
--tokens=tokens.txt
--lr=0.1
--gamma=0.8
--stepsize=30000
--maxgradnorm=0.7
--momentum=0.5
--mfsc=true
--batchsize=20
--iter=200
--lrcrit=0.002
--archdir=/data/ahnaf/wav2letter/dataset_prep/
--arch=network.arch
--enable_distributed=true
--criterion=ctc

log file: 
[001_log_200.zip](https://github.com/facebookresearch/wav2letter/files/3883663/001_log_200.zip)

**Comment by samin9796** [comment]
@tlikhomanenko Yes. Thank you.
